xdsl/interactive/app.py

```diff
@@ -432,6 +432,20 @@ def update_output_operation_count_tuple(self) -> None:
             # calculate diff of output and  input if there is one
             self.output_operation_count_datatable.add_row(k, v)
 
+            input_op_count_dict = dict(self.input_operation_count_tuple)
+            if k in input_op_count_dict:
+                diff = v - input_op_count_dict[k]
+                self.output_operation_count_datatable.add_row(k, v, diff)
+            else:
+                self.output_operation_count_datatable.add_row(k, v, "-")
+
+            # for k_input, v_input in self.input_operation_count_tuple:
+            #     if k == k_input:
+            #         diff = v - v_input
+            #         self.output_operation_count_datatable.add_row(k, v, diff)
+            #     else:
+            #         self.output_operation_count_datatable.add_row(k, v, "-")
+
```

Comment:
Have provided 2 different implementations - to loop through tuple to check if k is in the tuple (double for loop seemed a bit costly though...) so I thought might be cheaper to convert to dictionary and then do dictionary lookup? lmk which is preferred.

Reply:
I would recommend a third option, adding the diff to the data source tuple, since we have both the output and input counts at that stage, so the view calculation is a dumb display of data

Reply:
I don't quite understand what you mean. What is the data source tuple? We have the input count tuple and the output count tuple. Which one are you referring to?

Reply:
I mean the output count tuple